libcudacxx/include/cuda/std/__utility/declval.h

```diff
@@ -20,8 +20,25 @@
 #  pragma system_header
 #endif // no system header
 
+#include <cuda/std/__type_traits/void_t.h>
+
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
+// When variable templates are available, a faster implementation of declval
+// is available. It compiles approximately 2x faster than the fallback.
+#if !defined(_CCCL_NO_VARIABLE_TEMPLATES)
```

Comment:
I believe this is only possible with inline variables. Otherwise you get the dreaded duplicated symbol warning
```suggestion
#if !defined(_CCCL_NO_INLINE_VARIABLES)
```

Reply:
Global constants have internal linkage by default, which avoids those like errors.